mirrord/kube/src/resolved/cron_job.rs

```diff
@@ -0,0 +1,48 @@
+use std::collections::BTreeMap;
+
+use k8s_openapi::api::batch::v1::CronJob;
+
+use super::ResolvedResource;
+use crate::{
+    api::runtime::RuntimeDataFromLabels,
+    error::{KubeApiError, Result},
+};
+
+#[derive(Debug, Clone)]
+pub struct ResolvedCronJob {
+    pub resource: CronJob,
+    pub container: Option<String>,
+}
+
+impl RuntimeDataFromLabels for ResolvedResource<CronJob> {
+    type Resource = CronJob;
+
+    fn name(&self) -> &str {
+        self.resource.metadata.name.as_ref().unwrap().as_str()
```

Comment:
Is this code used or could be used by the operator? If so, maybe let's not `unwrap`, even though k8s is promises to populate that field? Maybe just change to `unwrap_or_default`? That way in the highly unlikely case that we get invalid information from k8s, the operator does not panic, but just has 1 session with this wrong name. Of course we could make this method return a `Result` or an `Option`, but IMO the complication of the interface is not worth if for the tiny chance this would happen. Up to you.

Reply:
`Cow` to the rescue, changed to `unwrap_or_default`. Had to change the return though, as we can't return a temporary (the _default).